### PR TITLE
Fix PubSub utility stuck in infinite loop when Redis connection is interrupted

### PR DESCRIPTION
**Pull request by masipcat**

### CODE REVIEW

File: guillotina/contrib/pubsub/utility.py [hunk 1]
```diff
@@ -58,6 +58,12 @@ async def finalize(self, app):
     async def real_subscribe(self, channel, channel_name):
         while channel_name in self._subscribers:
             try:
+                if channel is None:
+                    # wait 1 second and try to connect
+                    await asyncio.sleep(1)
```
Comment (vangheem) [inline guillotina/contrib/pubsub/utility.py:63]: Hhhm. This means I'm startup now there will always be a 1 second delay before this starts.

Comment (vangheem) [reply guillotina/contrib/pubsub/utility.py:63]: Should the sheep still be in exception handler but we just remove the finally block like you did?

**Review by vangheem — APPROVED** [review]
My only comment is that now startup is delayed for that

File: guillotina/contrib/pubsub/utility.py (hunk 1 above)
Comment (masipcat) [reply guillotina/contrib/pubsub/utility.py:63]: > Hhhm. This means I'm startup now there will always be a 1 second delay before this starts.

No, the channel is initialized here:

https://github.com/plone/guillotina/blob/98ee68b1c3c1a09bc5f65f9fc43d4e370a55843d/guillotina/contrib/pubsub/utility.py#L96-L97

Comment (masipcat) [reply guillotina/contrib/pubsub/utility.py:63]: > Should the sheep still be in exception handler but we just remove the finally block like you did?

I'm not 100% sure but the problem was that after catching the ConnectionError, the next time wasn't crashing, just iterating the empty async generator thus not executing the sleep inside the exception and producing the cpu spinning.

Comment (vangheem) [reply guillotina/contrib/pubsub/utility.py:63]: Ahh, I gotcha. That's a bit confusing.

**Pull request merged**